subprojects/docs/src/docs/userguide/what_is_gradle.adoc

```diff
@@ -12,133 +12,114 @@
+in every build. However, the configuration cache is only compatible with certain syntax.
 
-It's easy to view Gradle's build scripts as executable code, because that's what they are. But that's an implementation detail: well-designed build scripts describe _what_ steps are needed to build the software, not _how_ those steps should do the work. That's a job for custom task types and plugins.
+=== Build Cache
 
-[NOTE]
-====
-There is a common misconception that Gradle's power and flexibility come from the fact that its build scripts are code. This couldn't be further from the truth. It's the underlying model and API that provide the power. As we recommend in our best practices, you should <<authoring_maintainable_build_scripts#sec:avoid_imperative_logic_in_scripts,avoid putting much, if any, imperative logic in your build scripts>>.
-====
+Every time you run a build, Gradle runs a set of tasks.
+The build cache stores the inputs provided to and outputs generated by those tasks.
+When you run Gradle with the build cache enabled, Gradle checks the inputs for each
+task. If the build cache already contains output for a previously completed task's inputs, Gradle reuses that output immediately, without re-running the task.
+Gradle can speed up builds significantly by pulling data from the cache instead of running the same task repeatedly.
+However, the build cache requires annotated inputs and outputs on tasks.
 
-Yet there is one area in which it is useful to view a build script as executable code: in understanding how the syntax of the build script maps to Gradle's API. The API documentation — formed of the link:{groovyDslPath}/[Groovy DSL Reference] and the link:{javadocPath}/[Javadocs] — lists methods and properties, and refers to closures and actions. What do these mean within the context of a build script? Check out the <<groovy_build_script_primer#groovy_build_script_primer,Groovy Build Script Primer>> to learn the answer to that question so that you can make effective use of the API documentation.
+=== Script Design
 
-NOTE: As Gradle runs on the JVM, build scripts can also use the standard {javaApi}[Java API]. Groovy build scripts can additionally use the Groovy APIs, while Kotlin build scripts can use the Kotlin ones.
+Well-designed build scripts consist mostly of
+<<authoring_maintainable_build_scripts#sec:avoid_imperative_logic_in_scripts,declarative configuration, rather than imperative logic>>.
+That configuration runs during the configuration phase.
+Many builds also register instances of custom task types providing additional actions that run during the execution phase.
 
+It's easy to view Gradle's build scripts as executable code, because that's what they are.
+But that's an implementation detail: well-designed build scripts describe _what_ steps build the software,
+not _how_ those steps do the work. That's a job for custom task types and plugins.
```

Comment:
This section seems out of place from the other "Optimizations"

I'm also not sure what we're trying to convey in this section. I like that we're talking about declarative build scripts and how tasks/plugins should hide the implementation details, but I think that message is muddled here. 

```suggestion
Well-designed build scripts consist mostly of
<<authoring_maintainable_build_scripts#sec:avoid_imperative_logic_in_scripts,declarative configuration, rather than imperative logic>>.
That configuration runs during the configuration phase.
Many builds also register instances of custom task types providing additional actions that run during the execution phase.

It's easy to view Gradle's build scripts as executable code, because that's what they are.
But that's an implementation detail: well-designed build scripts describe _what_ steps build the software,
not _how_ those steps do the work. That's a job for custom task types and plugins.
```

Reply:
Would it make sense to bump this to the top level and rename it "best practices" or something similar? The idea of declarative build scripts is so core to modern idiomatic Gradle that I think it's worth bringing up, but I think you're correct -- it's not really an optimization, it's just the way you should write Gradle logic going forward.